create_db.py

```diff
@@ -0,0 +1,265 @@
+		    		elif pronoun == "den" and case == "Acc":
+		    			gender = "Masc"
+
+			    	if gender and spacy_gender:
+			    		count = count + 1
+			    		if gender == spacy_gender[0]:
+			    			correct = correct + 1
+			    			seed[gender].add(text)
+
+	print(f'Total nouns where a gender was assigned and Spacy has the gender categorised is {count}')
+	print(f'Total number of nouns correctly labelled (considering spacy gender categorisation as gold data)is {correct}. Hence, accuracy is {(correct/count)*100}')
+	return seed
+
+
+def create_suffix_bank(seed):
+	suffixes_masc = {}
+	suffixes_fem = {}
+	suffixes_neut = {}
+
+	# Masculine
+	c = 0
+	# loop through keys and create suffixes for each noun. only add suffixes of length less than or equal to 3
+	for nouns in set(seed["Masc"]):
+		suffixes = [nouns[i:] for i in range(len(nouns) - 3, len(nouns))]
+		for s in suffixes:
+			if s in suffixes_masc.keys():
+				suffixes_masc[s] = suffixes_masc[s] + 1
+			else:
+				suffixes_masc[s] = 1
+
+	# delete suffixes with frequency less than 3. Generate probability distribution for rest of the suffixes
+	for key in list(suffixes_masc.keys()):
+		if suffixes_masc[key] < 3:
+			del suffixes_masc[key]
+		else:
+			c = c + suffixes_masc[key]
+
+	for key, val in suffixes_masc.items():
+		suffixes_masc[key] = suffixes_masc[key]/c
+
+	# Feminine
+	c = 0
+	# loop through keys and create suffixes for each noun. only add suffixes of length less than or equal to 3
+	for nouns in set(seed["Fem"]):
+		suffixes = [nouns[i:] for i in range(len(nouns) - 3, len(nouns))]
+		for s in suffixes:
+			if s in suffixes_fem.keys():
+				suffixes_fem[s] = suffixes_fem[s] + 1
+			else:
+				suffixes_fem[s] = 1
+
+	# delete suffixes with frequency less than 3. Generate probability distribution for rest of the suffixes
+	for key in list(suffixes_fem.keys()):
+		if suffixes_fem[key] < 3:
+			del suffixes_fem[key]
+		else:
+			c = c + suffixes_fem[key]
+
+	for key, val in suffixes_fem.items():
+		suffixes_fem[key] = suffixes_fem[key]/c
+
+	# Neutral
+	c = 0
+	# loop through keys and create suffixes for each noun. only add suffixes of length less than or equal to 3
+	for nouns in set(seed["Neut"]):
+		suffixes = [nouns[i:] for i in range(len(nouns) - 3, len(nouns))]
+		for s in suffixes:
+			if s in suffixes_neut.keys():
+				suffixes_neut[s] = suffixes_neut[s] + 1
+			else:
+				suffixes_neut[s] = 1
+
+	# delete suffixes with frequency less than 3. Generate probability distribution for rest of the suffixes
+	for key in list(suffixes_neut.keys()):
+		if suffixes_neut[key] < 3:
+			del suffixes_neut[key]
+		else:
+			c = c + suffixes_neut[key]
+
+	for key, val in suffixes_neut.items():
+		suffixes_neut[key] = suffixes_neut[key]/c
+
+
+	return suffixes_masc, suffixes_fem, suffixes_neut
+
+def morphology_based_classification(corpus, seed, suffixes_masc, suffixes_fem, suffixes_neut):
+	# total number of nouns for which there was a calculated gender and a Spacy assigned gender
```

Comment:
Separate the actual classification and evaluation. This way it is hard to see where is the boundary.

Reply:
I see you did not do this. I will accept this for the purpose of the class, but:

In real world, you do classification and evaluation separately. Imagine, you are running this as a product for some company 
- the classification is what your customers pay you for (they send you a noun and you return the gender) and 
- evaluation is what you run internally so that you can trust your product API (alongside of unit test, integration tests, etc)

Reply:
I think I have implemented this, In terms of classification, `create_db.py` is the evaluation file, it creates the "suffix bank" which is a collection of suffixes for every gender. This collection is then written to a file after pickling. It also calculates how well the suffix bank works with respect to the nouns already present in the corpus which is where the accuracy scores come in. 

In terms of evaluation, `app.py` is the file that actually uses the suffix bank(and the suffix pickle files for every gender)  and predicts the gender of a noun when given as input. It also has an API call to a german gender noun API to compare. All we need to provide a potential customer would be the three gender `.pkl` files and `app.py`. 

Is this what you are looking for?